src/mm/allocator/mod.rs

```diff
@@ -1,54 +1,30 @@
 //! Implementation of the HermitCore Allocator for dynamically allocating heap memory
 //! in the kernel.
 
-use core::alloc::{AllocError, GlobalAlloc, Layout};
+mod bootstrap;
+mod bump;
+
+use core::alloc::{AllocError, Allocator, GlobalAlloc, Layout};
 use core::ptr;
 use core::ptr::NonNull;
 
 use align_address::Align;
 use hermit_sync::InterruptTicketMutex;
 use linked_list_allocator::Heap;
 
-use crate::mm::kernel_end_address;
+use self::bootstrap::BootstrapAllocator;
+use self::bump::BumpAllocator;
 use crate::HW_DESTRUCTIVE_INTERFERENCE_SIZE;
 
-struct BootstrapAllocator {
-	first_block: [u8; Self::SIZE],
-	index: usize,
-}
-
-impl BootstrapAllocator {
-	const SIZE: usize = 4096;
-
-	const fn new() -> Self {
-		Self {
-			first_block: [0xCC; Self::SIZE],
-			index: 0,
-		}
-	}
-
-	/// An allocation using the always available Bootstrap Allocator.
-	unsafe fn alloc(&mut self, layout: Layout) -> Result<NonNull<u8>, AllocError> {
-		let ptr = &mut self.first_block[self.index] as *mut u8;
-
-		self.index += layout.size();
-		if self.index >= Self::SIZE {
-			Err(AllocError)
-		} else {
-			Ok(NonNull::new(ptr).unwrap())
-		}
-	}
-}
-
 struct GlobalAllocator {
```

Comment:
A short doc comment stating why we have `bootstrap_allocator` as well as `heap` would be nice

Reply:
And stating what the initialization should look like would also be helpful.

Reply:
Thank you, this is actually very helpful now! :+1: